Dockerfile

```diff
@@ -16,15 +17,37 @@ RUN sed -ri "s/(httpredir|deb).debian.org/${APT_MIRROR:-deb.debian.org}/g" /etc/
  && sed -ri "s/(security).debian.org/${APT_MIRROR:-security.debian.org}/g" /etc/apt/sources.list
 ENV GO111MODULE=off
 
+# criu
+FROM base AS criu-src
+WORKDIR /usr/src/criu
+RUN git init . && git remote add origin "https://github.com/checkpoint-restore/criu.git"
+ARG CRIU_VERSION
+RUN git fetch --depth 1 origin "${CRIU_VERSION}" && git checkout -q FETCH_HEAD
```

Comment:
Curious why you're using a separate stage for this. Was this related to cross compiling (e.g.)? Otherwise, we may as well make it a `RUN` in the `criu` stage below

Would be nice if `ADD` could also extract, because then if would be just a

```dockerfile
ADD https://github.com/checkpoint-restore/criu/archive/refs/tags/${CRIU_VERSION}.tar.gz .
```

Reply:
Yes was related to cross comp with `TARGETPLATFORM` arg but not needed in this case.